langchain/retrievers/pinecone_hybrid_search.py

```diff
@@ -69,38 +66,10 @@ def create_index(
         index.upsert(vectors)
 
 
-def generate_sparse_vectors(context_batch: List[str], tokenizer: Any) -> List[Dict]:
-    # create batch of input_ids
-    inputs = tokenizer(
-        context_batch,
-        padding=True,
-        truncation=True,
-        max_length=512,  # special_tokens=False
-    )["input_ids"]
-    # create sparse dictionaries
-    sparse_embeds = build_dict(inputs)
-    return sparse_embeds
-
-
-def hybrid_scale(
-    dense: List[float], sparse: Dict, alpha: float
-) -> Tuple[List[float], Dict]:
-    # check alpha value is in range
-    if alpha < 0 or alpha > 1:
-        raise ValueError("Alpha must be between 0 and 1")
-    # scale sparse and dense vectors to create hybrid search vecs
-    hsparse = {
-        "indices": sparse["indices"],
-        "values": [v * (1 - alpha) for v in sparse["values"]],
-    }
-    hdense = [v * alpha for v in dense]
-    return hdense, hsparse
-
-
 class PineconeHybridSearchRetriever(BaseRetriever, BaseModel):
     embeddings: Embeddings
+    sparse_encoder: BaseSparseEncoder
```

Comment:
@hwchase17  - we kept it as separate encoder, but it might make sense to add functionalists to `Embeddings` class: `sparse_embed_query`, `sparse_embed_document` ... 

WDYT?

Reply:
imo would prefer to get one or two more instances of sparse encoders in the library before abstracting